src/services/services.ts

```diff
@@ -2599,24 +2585,18 @@ module ts {
                     /// TODO filter meaning based on the current context
                     let scopeNode = getScopeNode(previousToken, position, sourceFile);
                     let symbolMeanings = SymbolFlags.Type | SymbolFlags.Value | SymbolFlags.Namespace | SymbolFlags.Alias;
-                    let symbols = typeInfoResolver.getSymbolsInScope(scopeNode, symbolMeanings);
-
-                    getCompletionEntriesFromSymbols(symbols, activeCompletionSession);
+                    
+                    // Filter down to the symbol that matches the symbolName if we were given one.
+                    let predicate: (s: Symbol) => boolean = symbolName 

```

Comment:
Nope.  We only create the item if it has a name we can display:

            if (!displayName) {
                return undefined;
            }

         -- Cyrus

From: Daniel Rosenwasser [mailto:notifications@github.com]
Sent: Tuesday, March 24, 2015 12:34 AM
To: Microsoft/TypeScript
Cc: Cyrus Najmabadi
Subject: Re: [TypeScript] Share code between getCompletionsAtPosition getCompletionEntryDetails. (#2475)


In src/services/services.ts<https://github.com/Microsoft/TypeScript/pull/2475#discussion_r27007308>:

> @@ -2599,24 +2585,18 @@ module ts {

>                      /// TODO filter meaning based on the current context

>                      let scopeNode = getScopeNode(previousToken, position, sourceFile);

>                      let symbolMeanings = SymbolFlags.Type | SymbolFlags.Value | SymbolFlags.Namespace | SymbolFlags.Alias;

> -                    let symbols = typeInfoResolver.getSymbolsInScope(scopeNode, symbolMeanings);

> -

> -                    getCompletionEntriesFromSymbols(symbols, activeCompletionSession);

> +

> +                    // Filter down to the symbol that matches the symbolName if we were given one.

> +                    let predicate: (s: Symbol) => boolean = symbolName

Is it possible to have a symbol with no name?

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/2475/files#r27007308>.